### ISSUE TITLE
FileNotFoundError: [Errno 2] No such file or directory: 'webcam/webcam/settings.py.tmpl'

### ISSUE THREAD

**Issue by dfriestedt** [scrapy/scrapy#4281]
### Description

when I issue the following command

`scrapy startproject webcam`

I get the error message:

```
vagrant@vagrant:/vagrant$ scrapy startproject webcam
Traceback (most recent call last):
  File "/usr/local/bin/scrapy", line 11, in <module>
    sys.exit(execute())
  File "/usr/local/lib/python3.6/dist-packages/scrapy/cmdline.py", line 146, in execute
    _run_print_help(parser, _run_command, cmd, args, opts)
  File "/usr/local/lib/python3.6/dist-packages/scrapy/cmdline.py", line 100, in _run_print_help
    func(*a, **kw)
  File "/usr/local/lib/python3.6/dist-packages/scrapy/cmdline.py", line 154, in _run_command
    cmd.run(args, opts)
  File "/usr/local/lib/python3.6/dist-packages/scrapy/commands/startproject.py", line 109, in run
    ProjectName=string_camelcase(project_name))
  File "/usr/local/lib/python3.6/dist-packages/scrapy/utils/template.py", line 9, in render_templatefile
    with open(path, 'rb') as fp:
FileNotFoundError: [Errno 2] No such file or directory: 'webcam/webcam/settings.py.tmpl'
```

### Steps to Reproduce

1. Create a fresh vagrant machine with ubuntu 18.04 on a mac 10.14.6
2. Install scrapy per these instruction with all the dependancies: https://docs.scrapy.org/en/latest/intro/install.html
3. issue the command scrapy startproject webcam

**Expected behavior:** 
I expect it to create the project structure without errors.

**Actual behavior:** 
It creates the folder structure as expected BUT produces the error message in the description.

**Reproduces how often:** 
This happens on a fresh install.  on a new Ubuntu 18.04 box.  I tried to uninstall and reinstall scrapy.  It produces the same error.

### Versions

```
vagrant@vagrant:/vagrant$ pip show scrapy
Name: Scrapy
Version: 1.8.0
Summary: A high-level Web Crawling and Web Scraping framework
Home-page: https://scrapy.org
Author: Scrapy developers
Author-email: None
License: BSD
Location: /home/vagrant/.local/lib/python2.7/site-packages
Requires: pyOpenSSL, protego, service-identity, cryptography, Twisted, lxml, six, parsel, cssselect, w3lib, zope.interface, PyDispatcher, queuelib
```

### Additional context

This issue was identified here:  https://github.com/scrapy/scrapy/issues/4019.  I provided the same information above but that issue is closed.  I'm opening a new ticket here.

**Comment by elacuesta** [comment]
Could you post the result of `scrapy version -v`? Does the issue appear if you install Scrapy inside a [virtual environment?](https://docs.python.org/3/library/venv.html). What seems odd to me is that `pip show` says `Location: /home/vagrant/.local/lib/python2.7/site-packages` but the traceback says `/usr/local/lib/python3.6/dist-packages/scrapy/utils/template.py` (Python 2 vs 3).

**Comment by dfriestedt** [comment]
```
vagrant@vagrant:/vagrant$ scrapy version -v
Scrapy       : 1.8.0
lxml         : 4.4.2.0
libxml2      : 2.9.9
cssselect    : 1.1.0
parsel       : 1.5.2
w3lib        : 1.21.0
Twisted      : 19.10.0
Python       : 2.7.17 (default, Nov  7 2019, 10:07:09) - [GCC 7.4.0]
pyOpenSSL    : 19.1.0 (OpenSSL 1.1.1d  10 Sep 2019)
cryptography : 2.8
Platform     : Linux-4.15.0-72-generic-x86_64-with-Ubuntu-18.04-bionic
```
Should I try to install scrapy inside a virtual environment inside the vagrant box?  

I found the location  strange too.  No idea how it was installed in that location.  I'll create a new Vagrant machine and track the steps I'm making to install scrapy exactly.

**Comment by dfriestedt** [comment]
Here are my exact steps to recreate the issue

```
vagrant init bento/ubuntu-18.04
vagrant up
vagrant ssh
sudo apt update && sudo apt upgrade -y
sudo apt-get install python-dev python-pip libxml2-dev libxslt1-dev zlib1g-dev libffi-dev libssl-dev
sudo apt-get install python3 python3-dev
sudo pip install scrapy
```
Note the sudo to run `sudo pip install scrapy` .

When I try to install scrapy without the sudo it installs fine, but I get the following error when I run scrapy

```vagrant@vagrant:/vagrant$ scrapy startproject webcam
-bash: /usr/local/bin/scrapy: No such file or directory```

**Comment by dfriestedt** [comment]
I just created a virtual environment inside the vagrant box and got the same error.  Here are my steps

```
vagrant@vagrant:/vagrant$ sudo apt-get install python3-venv
vagrant@vagrant:/vagrant$ python3 -m venv ENV
vagrant@vagrant:/vagrant$ source ENV/bin/activate
(ENV) vagrant@vagrant:/vagrant$ pip install wheel
```
Note:  If I do not install wheel I get `error: invalid command 'bdist_wheel'` . But I know this is not a required step in the scrapy instructions

```
(ENV) vagrant@vagrant:/vagrant$ pip install scrapy
(ENV) vagrant@vagrant:/vagrant$ scrapy startproject webcam

Traceback (most recent call last):
  File "/vagrant/ENV/bin/scrapy", line 11, in <module>
    sys.exit(execute())
  File "/vagrant/ENV/lib/python3.6/site-packages/scrapy/cmdline.py", line 146, in execute
    _run_print_help(parser, _run_command, cmd, args, opts)
  File "/vagrant/ENV/lib/python3.6/site-packages/scrapy/cmdline.py", line 100, in _run_print_help
    func(*a, **kw)
  File "/vagrant/ENV/lib/python3.6/site-packages/scrapy/cmdline.py", line 154, in _run_command
    cmd.run(args, opts)
  File "/vagrant/ENV/lib/python3.6/site-packages/scrapy/commands/startproject.py", line 109, in run
    ProjectName=string_camelcase(project_name))
  File "/vagrant/ENV/lib/python3.6/site-packages/scrapy/utils/template.py", line 9, in render_templatefile
    with open(path, 'rb') as fp:
FileNotFoundError: [Errno 2] No such file or directory: 'webcam/webcam/settings.py.tmpl'
```

**Comment by elacuesta** [comment]
I think this is a Vagrant-specific issue, related to file system permissions and the way folders are shared between the host and the virtual machine. The issue only appears for me while working on the `/vagrant` directory, which AFAICT is a mount point for the directory which holds the `Vagrantfile` in the host system. I tried on a macOS and a Linux Mint machine, and the issue appeared both times.

It all works fine while working on the user's home directory. With a fresh Vagrant installation (2.2.6) I tried both `bento/ubuntu-18.04` and `ubuntu/bionic64`, and got the same results:

```
# host machine
vagrant init bento/ubuntu-18.04
vagrant up
vagrant ssh
```

```
# vagrant machine
sudo apt-get update
sudo apt-get install python3-venv
python3 -m venv virtualenv
source virtualenv/bin/activate
pip install wheel
pip install scrapy
scrapy startproject webcam
scrapy version -v
pip show scrapy
```


```
vagrant@vagrant:~$ sudo apt-get update
Hit:1 http://archive.ubuntu.com/ubuntu bionic InRelease
Get:2 http://security.ubuntu.com/ubuntu bionic-security InRelease [88.7 kB]
Get:3 http://archive.ubuntu.com/ubuntu bionic-updates InRelease [88.7 kB]
Get:4 http://security.ubuntu.com/ubuntu bionic-security/main amd64 Packages [611 kB]
Get:5 http://archive.ubuntu.com/ubuntu bionic-backports InRelease [74.6 kB]
Get:6 http://archive.ubuntu.com/ubuntu bionic-updates/main i386 Packages [632 kB]
Get:7 http://security.ubuntu.com/ubuntu bionic-security/main i386 Packages [422 kB]
Get:8 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 Packages [836 kB]
Get:9 http://security.ubuntu.com/ubuntu bionic-security/main Translation-en [200 kB]
Get:10 http://security.ubuntu.com/ubuntu bionic-security/restricted amd64 Packages [18.4 kB]
Get:11 http://security.ubuntu.com/ubuntu bionic-security/restricted Translation-en [5,408 B]
Get:12 http://security.ubuntu.com/ubuntu bionic-security/universe i386 Packages [608 kB]
Get:13 http://archive.ubuntu.com/ubuntu bionic-updates/main Translation-en [293 kB]
Get:14 http://security.ubuntu.com/ubuntu bionic-security/universe amd64 Packages [634 kB]
Get:15 http://archive.ubuntu.com/ubuntu bionic-updates/restricted amd64 Packages [28.2 kB]
Get:16 http://archive.ubuntu.com/ubuntu bionic-updates/restricted i386 Packages [9,656 B]
Get:17 http://archive.ubuntu.com/ubuntu bionic-updates/restricted Translation-en [7,504 B]
Get:18 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 Packages [1,045 kB]
Get:19 http://security.ubuntu.com/ubuntu bionic-security/universe Translation-en [213 kB]
Get:20 http://security.ubuntu.com/ubuntu bionic-security/multiverse amd64 Packages [6,116 B]
Get:21 http://security.ubuntu.com/ubuntu bionic-security/multiverse i386 Packages [4,284 B]
Get:22 http://archive.ubuntu.com/ubuntu bionic-updates/universe i386 Packages [1,004 kB]
Get:23 http://archive.ubuntu.com/ubuntu bionic-updates/universe Translation-en [322 kB]
Get:24 http://archive.ubuntu.com/ubuntu bionic-updates/multiverse i386 Packages [7,496 B]
Get:25 http://archive.ubuntu.com/ubuntu bionic-updates/multiverse amd64 Packages [9,512 B]
Get:26 http://archive.ubuntu.com/ubuntu bionic-updates/multiverse Translation-en [4,540 B]
Get:27 http://archive.ubuntu.com/ubuntu bionic-backports/universe amd64 Packages [4,032 B]
Get:28 http://archive.ubuntu.com/ubuntu bionic-backports/universe i386 Packages [4,024 B]
Get:29 http://archive.ubuntu.com/ubuntu bionic-backports/universe Translation-en [1,900 B]
Fetched 7,183 kB in 4s (1,602 kB/s)
Reading package lists... Done

vagrant@vagrant:~$ sudo apt-get install python3-venv
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following additional packages will be installed:
  python-pip-whl python3-distutils python3-lib2to3 python3.6-venv
The following NEW packages will be installed:
  python-pip-whl python3-distutils python3-lib2to3 python3-venv python3.6-venv
0 upgraded, 5 newly installed, 0 to remove and 28 not upgraded.
Need to get 1,882 kB of archives.
After this operation, 5,043 kB of additional disk space will be used.
Do you want to continue? [Y/n] y
Get:1 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 python-pip-whl all 9.0.1-2.3~ubuntu1.18.04.1 [1,653 kB]
Get:2 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 python3-lib2to3 all 3.6.9-1~18.04 [77.4 kB]
Get:3 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 python3-distutils all 3.6.9-1~18.04 [144 kB]
Get:4 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 python3.6-venv amd64 3.6.9-1~18.04 [6,180 B]
Get:5 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 python3-venv amd64 3.6.7-1~18.04 [1,208 B]
Fetched 1,882 kB in 3s (633 kB/s)
Selecting previously unselected package python-pip-whl.
(Reading database ... 42631 files and directories currently installed.)
Preparing to unpack .../python-pip-whl_9.0.1-2.3~ubuntu1.18.04.1_all.deb ...
Unpacking python-pip-whl (9.0.1-2.3~ubuntu1.18.04.1) ...
Selecting previously unselected package python3-lib2to3.
Preparing to unpack .../python3-lib2to3_3.6.9-1~18.04_all.deb ...
Unpacking python3-lib2to3 (3.6.9-1~18.04) ...
Selecting previously unselected package python3-distutils.
Preparing to unpack .../python3-distutils_3.6.9-1~18.04_all.deb ...
Unpacking python3-distutils (3.6.9-1~18.04) ...
Selecting previously unselected package python3.6-venv.
Preparing to unpack .../python3.6-venv_3.6.9-1~18.04_amd64.deb ...
Unpacking python3.6-venv (3.6.9-1~18.04) ...
Selecting previously unselected package python3-venv.
Preparing to unpack .../python3-venv_3.6.7-1~18.04_amd64.deb ...
Unpacking python3-venv (3.6.7-1~18.04) ...
Setting up python-pip-whl (9.0.1-2.3~ubuntu1.18.04.1) ...
Setting up python3.6-venv (3.6.9-1~18.04) ...
Setting up python3-lib2to3 (3.6.9-1~18.04) ...
Setting up python3-distutils (3.6.9-1~18.04) ...
Setting up python3-venv (3.6.7-1~18.04) ...
Processing triggers for man-db (2.8.3-2ubuntu0.1) ...

vagrant@vagrant:~$ python3 -m venv virtualenv

vagrant@vagrant:~$ source virtualenv/bin/activate

(virtualenv) vagrant@vagrant:~$ pip install wheel
Collecting wheel
  Downloading https://files.pythonhosted.org/packages/58/2b/2a307602a01ade0a4ef1774213f978aac87d1f4b9a875942c8cead888871/wheel-0.34.0-py2.py3-none-any.whl
Installing collected packages: wheel
Successfully installed wheel-0.34.0

(virtualenv) vagrant@vagrant:~$ pip install scrapy
Collecting scrapy
  Downloading https://files.pythonhosted.org/packages/3b/e4/69b87d7827abf03dea2ea984230d50f347b00a7a3897bc93f6ec3dafa494/Scrapy-1.8.0-py2.py3-none-any.whl (238kB)
    100% |████████████████████████████████| 245kB 860kB/s
Collecting PyDispatcher>=2.0.5 (from scrapy)
  Downloading https://files.pythonhosted.org/packages/cd/37/39aca520918ce1935bea9c356bcbb7ed7e52ad4e31bff9b943dfc8e7115b/PyDispatcher-2.0.5.tar.gz
Collecting cryptography>=2.0 (from scrapy)
  Downloading https://files.pythonhosted.org/packages/45/73/d18a8884de8bffdcda475728008b5b13be7fbef40a2acc81a0d5d524175d/cryptography-2.8-cp34-abi3-manylinux1_x86_64.whl (2.3MB)
    100% |████████████████████████████████| 2.3MB 449kB/s
Collecting service-identity>=16.0.0 (from scrapy)
  Downloading https://files.pythonhosted.org/packages/e9/7c/2195b890023e098f9618d43ebc337d83c8b38d414326685339eb024db2f6/service_identity-18.1.0-py2.py3-none-any.whl
Collecting cssselect>=0.9.1 (from scrapy)
  Downloading https://files.pythonhosted.org/packages/3b/d4/3b5c17f00cce85b9a1e6f91096e1cc8e8ede2e1be8e96b87ce1ed09e92c5/cssselect-1.1.0-py2.py3-none-any.whl
Collecting parsel>=1.5.0 (from scrapy)
  Downloading https://files.pythonhosted.org/packages/86/c8/fc5a2f9376066905dfcca334da2a25842aedfda142c0424722e7c497798b/parsel-1.5.2-py2.py3-none-any.whl
Collecting w3lib>=1.17.0 (from scrapy)
  Downloading https://files.pythonhosted.org/packages/6a/45/1ba17c50a0bb16bd950c9c2b92ec60d40c8ebda9f3371ae4230c437120b6/w3lib-1.21.0-py2.py3-none-any.whl
Collecting Twisted>=17.9.0; python_version >= "3.5" (from scrapy)
  Downloading https://files.pythonhosted.org/packages/88/e2/0c21fadf0dff02d145db02f24a6ed2c24993e7242d138babbca41de2f5a2/Twisted-19.10.0-cp36-cp36m-manylinux1_x86_64.whl (3.1MB)
    100% |████████████████████████████████| 3.1MB 463kB/s
Collecting zope.interface>=4.1.3 (from scrapy)
  Downloading https://files.pythonhosted.org/packages/16/1c/d9e4d1e4eb9777ae675c5ac01290e70012498944d5e743bd2777d1096ad7/zope.interface-4.7.1-cp36-cp36m-manylinux1_x86_64.whl (168kB)
    100% |████████████████████████████████| 174kB 3.3MB/s
Collecting queuelib>=1.4.2 (from scrapy)
  Downloading https://files.pythonhosted.org/packages/4c/85/ae64e9145f39dd6d14f8af3fa809a270ef3729f3b90b3c0cf5aa242ab0d4/queuelib-1.5.0-py2.py3-none-any.whl
Collecting six>=1.10.0 (from scrapy)
  Downloading https://files.pythonhosted.org/packages/65/eb/1f97cb97bfc2390a276969c6fae16075da282f5058082d4cb10c6c5c1dba/six-1.14.0-py2.py3-none-any.whl
Collecting pyOpenSSL>=16.2.0 (from scrapy)
  Downloading https://files.pythonhosted.org/packages/9e/de/f8342b68fa9e981d348039954657bdf681b2ab93de27443be51865ffa310/pyOpenSSL-19.1.0-py2.py3-none-any.whl (53kB)
    100% |████████████████████████████████| 61kB 3.9MB/s
Collecting lxml>=3.5.0 (from scrapy)
  Downloading https://files.pythonhosted.org/packages/68/30/affd16b77edf9537f5be051905f33527021e20d563d013e8c42c7fd01949/lxml-4.4.2-cp36-cp36m-manylinux1_x86_64.whl (5.8MB)
    100% |████████████████████████████████| 5.8MB 241kB/s
Collecting protego>=0.1.15 (from scrapy)
  Downloading https://files.pythonhosted.org/packages/db/6e/bf6d5e4d7cf233b785719aaec2c38f027b9c2ed980a0015ec1a1cced4893/Protego-0.1.16.tar.gz (3.2MB)
    100% |████████████████████████████████| 3.2MB 407kB/s
Collecting cffi!=1.11.3,>=1.8 (from cryptography>=2.0->scrapy)
  Downloading https://files.pythonhosted.org/packages/49/72/0d42f94fe94afa8030350c26e9d787219f3f008ec9bf6b86c66532b29236/cffi-1.13.2-cp36-cp36m-manylinux1_x86_64.whl (397kB)
    100% |████████████████████████████████| 399kB 2.3MB/s
Collecting pyasn1 (from service-identity>=16.0.0->scrapy)
  Downloading https://files.pythonhosted.org/packages/62/1e/a94a8d635fa3ce4cfc7f506003548d0a2447ae76fd5ca53932970fe3053f/pyasn1-0.4.8-py2.py3-none-any.whl (77kB)
    100% |████████████████████████████████| 81kB 2.5MB/s
Collecting pyasn1-modules (from service-identity>=16.0.0->scrapy)
  Downloading https://files.pythonhosted.org/packages/95/de/214830a981892a3e286c3794f41ae67a4495df1108c3da8a9f62159b9a9d/pyasn1_modules-0.2.8-py2.py3-none-any.whl (155kB)
    100% |████████████████████████████████| 163kB 4.4MB/s
Collecting attrs>=16.0.0 (from service-identity>=16.0.0->scrapy)
  Downloading https://files.pythonhosted.org/packages/a2/db/4313ab3be961f7a763066401fb77f7748373b6094076ae2bda2806988af6/attrs-19.3.0-py2.py3-none-any.whl
Collecting incremental>=16.10.1 (from Twisted>=17.9.0; python_version >= "3.5"->scrapy)
  Downloading https://files.pythonhosted.org/packages/f5/1d/c98a587dc06e107115cf4a58b49de20b19222c83d75335a192052af4c4b7/incremental-17.5.0-py2.py3-none-any.whl
Collecting Automat>=0.3.0 (from Twisted>=17.9.0; python_version >= "3.5"->scrapy)
  Downloading https://files.pythonhosted.org/packages/e5/11/756922e977bb296a79ccf38e8d45cafee446733157d59bcd751d3aee57f5/Automat-0.8.0-py2.py3-none-any.whl
Collecting PyHamcrest>=1.9.0 (from Twisted>=17.9.0; python_version >= "3.5"->scrapy)
  Downloading https://files.pythonhosted.org/packages/ac/6c/a641af18e416e6501c10b03742387176626a1d48196100160df796f36632/PyHamcrest-2.0.0-py3-none-any.whl (51kB)
    100% |████████████████████████████████| 61kB 3.3MB/s
Collecting hyperlink>=17.1.1 (from Twisted>=17.9.0; python_version >= "3.5"->scrapy)
  Downloading https://files.pythonhosted.org/packages/7f/91/e916ca10a2de1cb7101a9b24da546fb90ee14629e23160086cf3361c4fb8/hyperlink-19.0.0-py2.py3-none-any.whl
Collecting constantly>=15.1 (from Twisted>=17.9.0; python_version >= "3.5"->scrapy)
  Downloading https://files.pythonhosted.org/packages/b9/65/48c1909d0c0aeae6c10213340ce682db01b48ea900a7d9fce7a7910ff318/constantly-15.1.0-py2.py3-none-any.whl
Requirement already satisfied: setuptools in ./virtualenv/lib/python3.6/site-packages (from zope.interface>=4.1.3->scrapy)
Collecting pycparser (from cffi!=1.11.3,>=1.8->cryptography>=2.0->scrapy)
  Downloading https://files.pythonhosted.org/packages/68/9e/49196946aee219aead1290e00d1e7fdeab8567783e83e1b9ab5585e6206a/pycparser-2.19.tar.gz (158kB)
    100% |████████████████████████████████| 163kB 3.6MB/s
Collecting idna>=2.5 (from hyperlink>=17.1.1->Twisted>=17.9.0; python_version >= "3.5"->scrapy)
  Downloading https://files.pythonhosted.org/packages/14/2c/cd551d81dbe15200be1cf41cd03869a46fe7226e7450af7a6545bfc474c9/idna-2.8-py2.py3-none-any.whl (58kB)
    100% |████████████████████████████████| 61kB 3.3MB/s
Building wheels for collected packages: PyDispatcher, protego, pycparser
  Running setup.py bdist_wheel for PyDispatcher ... done
  Stored in directory: /home/vagrant/.cache/pip/wheels/88/99/96/cfef6665f9cb1522ee6757ae5955feedf2fe25f1737f91fa7f
  Running setup.py bdist_wheel for protego ... done
  Stored in directory: /home/vagrant/.cache/pip/wheels/51/01/d1/4a2286a976dccd025ba679acacfe37320540df0f2283ecab12
  Running setup.py bdist_wheel for pycparser ... done
  Stored in directory: /home/vagrant/.cache/pip/wheels/f2/9a/90/de94f8556265ddc9d9c8b271b0f63e57b26fb1d67a45564511
Successfully built PyDispatcher protego pycparser
Installing collected packages: PyDispatcher, pycparser, cffi, six, cryptography, pyasn1, pyasn1-modules, attrs, service-identity, cssselect, lxml, w3lib, parsel, incremental, zope.interface, Automat, PyHamcrest, idna, hyperlink, constantly, Twisted, queuelib, pyOpenSSL, protego, scrapy
Successfully installed Automat-0.8.0 PyDispatcher-2.0.5 PyHamcrest-2.0.0 Twisted-19.10.0 attrs-19.3.0 cffi-1.13.2 constantly-15.1.0 cryptography-2.8 cssselect-1.1.0 hyperlink-19.0.0 idna-2.8 incremental-17.5.0 lxml-4.4.2 parsel-1.5.2 protego-0.1.16 pyOpenSSL-19.1.0 pyasn1-0.4.8 pyasn1-modules-0.2.8 pycparser-2.19 queuelib-1.5.0 scrapy-1.8.0 service-identity-18.1.0 six-1.14.0 w3lib-1.21.0 zope.interface-4.7.1

(virtualenv) vagrant@vagrant:~$ scrapy startproject webcam
New Scrapy project 'webcam', using template directory '/home/vagrant/virtualenv/lib/python3.6/site-packages/scrapy/templates/project', created in:
    /home/vagrant/webcam

You can start your first spider with:
    cd webcam
    scrapy genspider example example.com

(virtualenv) vagrant@vagrant:~$ scrapy version -v
Scrapy       : 1.8.0
lxml         : 4.4.2.0
libxml2      : 2.9.9
cssselect    : 1.1.0
parsel       : 1.5.2
w3lib        : 1.21.0
Twisted      : 19.10.0
Python       : 3.6.9 (default, Nov  7 2019, 10:44:02) - [GCC 8.3.0]
pyOpenSSL    : 19.1.0 (OpenSSL 1.1.1d  10 Sep 2019)
cryptography : 2.8
Platform     : Linux-4.15.0-72-generic-x86_64-with-Ubuntu-18.04-bionic

(virtualenv) vagrant@vagrant:~$ pip show scrapy
Name: Scrapy
Version: 1.8.0
Summary: A high-level Web Crawling and Web Scraping framework
Home-page: https://scrapy.org
Author: Scrapy developers
Author-email: None
License: BSD
Location: /home/vagrant/virtualenv/lib/python3.6/site-packages
Requires: six, Twisted, service-identity, cssselect, pyOpenSSL, zope.interface, queuelib, lxml, w3lib, protego, cryptography, parsel, PyDispatcher
```

**Comment by elacuesta** [comment]
The original issue (#4019) says "I use VBox and have share folder from Windows", I think it's safe to assume the cause of the issue is the same. I don't think we should (maybe even _can_) do anything on the Scrapy side to mitigate this.

**Comment by dfriestedt** [comment]
Understood and thank you for looking at this.  I moved to anaconda and that environment is working well.  I'll keep trouble shooting this on my end and post a solution if I ever figure it out.

**Comment by kumaxim** [comment]
I catch the same issue recently in my fresh Vagrant enviroment. 

I have Ubuntu 20.04.3 LTS on host and the same OS in guest. Here is very basic Vagrantfile that allow to reproduce the error:
```
Vagrant.configure("2") do |config|
  config.vm.define "ubuntu-desktop" do |c|
    c.vm.box = "peru/ubuntu-20.04-desktop-amd64"
    c.vm.box_version = "20210901.01"
    c.vm.box_check_update = false
  end

  config.vm.synced_folder "app/", "/home/vagrant/app",
    create: true,
    owner: "vagrant",
    group: "vagrant"

  config.vm.provider "virtualbox" do |vb|
    vb.gui = false
  end

  config.vm.provision "shell", inline: <<-SHELL
    apt-get install -y curl python3-dev python3-pip python3-venv python3-wheel
  SHELL
end
```

As @elacuesta supposed this is Vagrant-specific issue. He was not far from the truth. The problem is the filesystem that Vagrant use for VirtualBox machines by default. This is `vboxsf` and it's very slow in operation. 

If you'll create new Virtualbox machine based on Vagrantfile below, login into(`vagrant ssh`) and perform the following commands:
```
python3 -m venv /home/vagrant/app/venv
source /home/vagrant/app/venv/bin/activate
pip3 install wheel
pip3 install selenium scrapy

scrapy startproject webcam
```

...you'll get the error that @ngufyi listed in the first message. 

The solution would be change filesystem for shared folders. The simplies way is using `nfs` instead of `vboxfs`. 

First you should prepare both host and guest OS.

Install NFS packages on host:
`# apt-get install nfs-kernel-server nfs-common`

...and this one into guest OS:
`# apt-get install nfs-common`

After that you must change `Vagrantfile` in the following way:
```
Vagrant.configure("2") do |config|
   ...
  # Create a private network, which allows host-only access to the machine
  config.vm.network "private_network", type: "dhcp"

  # Change sync_folders options
  config.vm.synced_folder "app/", "/home/vagrant/app",
    create: true,
    type: "nfs"
end
```

Reload machine `vagrant reload` and confirm to export setting into `/etc/exports` when Vagrant ask you to confirm privileges. 
Login into machine again `vagrant ssh` and run `scrapy startproject webcam` into you Python's virtual enviroment. 

Now everything will be fine.



__P.S.:__ It was not easy to catch this error. I use PyCharm. Initially I set breakpoint in 9 line of `render_templatefile` function within `/home/vagrant/app/venv/lib/python3.8/dist-packages/scrapy/utils/template.py` file and run `scrapy startproject webcam` with pauses on this liine. I surprised when command will be completed successfully in debug mode. However, this gave me idea to put my attention into right field. 

I not sure whether Scapy's developers should do anything. I leave it to their discretion.

**Comment by UnimelbLachlan** [comment]
I run Linux on virtualbox (because I need a Windows host for work), and think that being unable to support  vboxfs  is a serious bug.  Could this issue be reopened, or should I create a new bug with kumaxim's explanation?

Note that the patch is trivial
[patch.txt](https://github.com/user-attachments/files/16027959/patch.txt)

--- template.py 2024-06-28 20:15:05.365195500 +1000
+++ /home/llandrew/.local/lib/python3.10/site-packages/scrapy/utils/template.py 2024-06-28 20:15:14.545109733 +1000
@@ -9,7 +9,13 @@
 
 def render_templatefile(path: Union[str, PathLike], **kwargs: Any) -> None:
     path_obj = Path(path)
-    raw = path_obj.read_text("utf8")
+    try :
+        raw = path_obj.read_text("utf8")
+    except FileNotFoundError :
+        # vboxfs is slow, causing this read to fail. Allow time and try again
+        import time
+        time.sleep(0.1)
+        raw = path_obj.read_text("utf8")
 
     content = string.Template(raw).substitute(**kwargs)

**Comment by Gallaecio** [comment]
That patch may be trivial, but if such a simple read fails, we would need to apply that to every read (and write?) operation there is, which sounds far from trivial.

Scrapy works on Windows, and there is WLS in Windows to run Linux without VirtualBox, so using VirtualBox should rarely be necessary. And if reading an existing file raises FileNotFoundError when using VirtualBox, it sounds like that should be fixed in VirtualBox, not in Scrapy.

**Comment by UnimelbLachlan** [comment]
Thanks for the prompt reply.  I agree that it should be fixed in VirtualBox, but that fix would presumably be much more difficult, and I don't expect Oracle to fix it.

I'm happy with you not applying the patch -- as you say, it is probably needed in other places which would become really ugly -- but at least it shows people what they need to do if they're in my position.

(WSL is OK for running code, but not great for development.)
